tests/tacacs/test_authorization.py

```diff
@@ -0,0 +1,342 @@
+            If command have local permission, user can run command.
+    """
+    stdout, stderr = ssh_run_command(ssh_client, "show aaa")
+    check_ssh_output(stdout, 'AAA authentication')
+
+    """
+        Verify TACACS+ user run command in server side whitelist:
+            If command not have local permission, user can't run command.
+    """
+    stdout, stderr = ssh_run_command(ssh_client, "config aaa")
+    check_ssh_output(stderr, 'Root privileges are required for this operation')
+
+    # Verify TACACS+ user can run command not in server side whitelist, but have local permission.
+    stdout, stderr = ssh_run_command(ssh_client, "cat /etc/passwd")
+    check_ssh_output(stdout, 'root:x:0:0:root:/root:/bin/bash')
+
+    # Verify Local user can't login.
+    check_ssh_connect_remote_failed(dutip, creds_all_duts[duthost]['tacacs_local_user'],
+                             creds_all_duts[duthost]['tacacs_local_user_passwd'])
+    
+    ssh_client.close()
+
+
+def test_authorization_tacacs_and_local_then_server_down_after_login(localhost, duthosts, enum_rand_one_per_hwsku_hostname, creds_all_duts,ptfhost, check_tacacs):
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+    duthost.shell("sudo config aaa authorization \"tacacs+ local\"")
+    setup_local_user(duthost, creds_all_duts)
+    dutip = duthost.host.options['inventory_manager'].get_host(duthost.hostname).vars['ansible_host']
+
+    # Create a ssh connection to sonic device
+    ssh_client = ssh_connect_remote(dutip, creds_all_duts[duthost]['tacacs_authorization_user'],
+                         creds_all_duts[duthost]['tacacs_authorization_user_passwd'])
+
+    # Shutdown tacacs server
+    stop_tacacs_server(ptfhost)
+    
+    # Verify TACACS+ user can run command not in server side whitelist but have permission in local.
+    stdout, stderr = ssh_run_command(ssh_client, "cat /etc/passwd")
+    check_ssh_output(stdout, 'root:x:0:0:root:/root:/bin/bash')
+    
+    # Verify TACACS+ user can't run command in server side whitelist also not have permission in local.
+    stdout, stderr = ssh_run_command(ssh_client, "config tacacs")
+    check_ssh_output(stdout, '/usr/local/bin/config not authorized by TACACS+ with given arguments, not executing')
+    check_ssh_output(stderr, 'Root privileges are required for this operation')
+
+    # Verify Local user can login, and run command with local permission.
+    ssh_client_local = ssh_connect_remote(dutip, creds_all_duts[duthost]['tacacs_local_user'],
+                         creds_all_duts[duthost]['tacacs_local_user_passwd'])
+
+    # Start tacacs server
+    start_tacacs_server(ptfhost)
+    
+    # Verify after Local user login, then server becomes accessible, Local user still can run command with local permission.
+    stdout, stderr = ssh_run_command(ssh_client_local, "show aaa")
+    check_ssh_output(stdout, 'AAA authentication')
+
+    # Cleanup
+    ssh_client.close()
+    ssh_client_local.close()
```

Comment:
Use a fixture [https://docs.pytest.org/en/6.2.x/fixture.html#teardown-cleanup-aka-fixture-finalization](https://docs.pytest.org/en/6.2.x/fixture.html#teardown-cleanup-aka-fixture-finalization)?

Reply:
Fixed.
And local_user_client is different with authorization_user_client, because local user can login after tacacs server down.